src/profile/helpers/updateMutationVariables.ts

```diff
@@ -0,0 +1,119 @@
+import { FormValues } from '../components/editProfileForm/EditProfileForm';
+import {
+  AddressType,
+  CreateEmailInput,
+  EmailType,
+  MyProfileQuery,
+  MyProfileQuery_myProfile_emails_edges_node as Email,
+  PhoneType,
+  UpdateEmailInput,
+} from '../../graphql/generatedTypes';
+
+const getAddress = (formValues: FormValues, profile?: MyProfileQuery) => {
+  if (profile?.myProfile?.primaryAddress?.id) {
+    return {
+      updateAddresses: [
+        {
+          id: profile.myProfile.primaryAddress.id,
+          address: formValues.address,
+          city: formValues.city,
+          postalCode: formValues.postalCode,
+          primary: true,
+          addressType: AddressType.OTHER,
+          countryCode: formValues.countryCode,
+        },
+      ],
+    };
+  }
+
+  return {
+    addAddresses: [
+      formValues.address || formValues.postalCode || formValues.city
+        ? {
+            address: formValues.address,
```

Comment:
Lot's of these rows are the same as compared to the first return. You've used the same pattern a few times in this file. Imo it makes maintenance a bit harder because you have potentially have more places you have to edit.

Reply:
Thats true. I tried to throw all of these fields to variable and use spread operator, that however results typing breaking (it can't find ID for some reason 😅 ). These methods are going to be replaced anyway when support for multiple addresses and phones are added.